### ISSUE TITLE
The behavior of EarlyStopping is not the same between 0.7.5 with 0.7.3

### ISSUE THREAD

**Issue by NickYi1990** [PyTorchLightning/pytorch-lightning#1699]
## 🐛 Bug

The behavior of EarlyStopping is not the same between 0.7.5 with 0.7.3, If i set patience=1, 0.7.3 will print two epoch when the second is worse than the first, but 0.7.5 will just print the first which is not consistent with the description of patience parameter

#### Code sample
```
early_stop_callback = EarlyStopping(monitor='valid_roc_auc', patience=0, mode='max', verbose=True)
# Instance Model, Trainer and train model
model = PlantModel(hparams)
trainer = pl.Trainer(gpus=[hparams.gpus],
                     max_epochs=hparams.max_epochs,
                     early_stop_callback=early_stop_callback,
                     progress_bar_refresh_rate=0,
                     num_sanity_val_steps=0,
                     profiler=True,
                     gradient_clip_val=hparams.gradient_clip_val)
trainer.fit(model)
print(trainer.early_stop_callback.wait)
```

### Environment

* CUDA:
        - GPU:
                - GeForce RTX 2080 Ti
                - GeForce RTX 2080 Ti
                - GeForce RTX 2080 Ti
                - GeForce RTX 2080 Ti
        - available:         True
        - version:           10.1
* Packages:
        - numpy:             1.18.1
        - pyTorch_debug:     False
        - pyTorch_version:   1.4.0
        - pytorch-lightning: 0.7.5
        - tensorboard:       2.2.1
        - tqdm:              4.42.1
* System:
        - OS:                Linux
        - architecture:
                - 64bit
                - 
        - processor:         x86_64
        - python:            3.7.6
        - version:           #38~18.04.1-Ubuntu SMP Tue Mar 31 04:17:56 UTC 2020

**Comment by williamFalcon** [comment]
@jeremyjordan

**Comment by jeremyjordan** [comment]
yes i believe this is the same issue reported [in this comment](https://github.com/PyTorchLightning/pytorch-lightning/issues/1464#issuecomment-619117767) and will be addressed in #1504

**Comment by NickYi1990** [comment]
> reported

Yes, exactly the same problem, It seems the only way to use 0.7.5 is to change the source code my self😢
